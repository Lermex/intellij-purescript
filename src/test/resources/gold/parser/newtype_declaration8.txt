Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Cool')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('c')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Cool')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PSASTWrapperElement(TypeConstructor)
                            PsiElement(proper name)('String')
                            PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PsiElement(()('(')
                        PSASTWrapperElement(Type)
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(TypeConstructor)
                              PSASTWrapperElement(Qualified)
                                PSASTWrapperElement(TypeConstructor)
                                  PsiElement(proper name)('Array')
                                  PsiElement(whitespace)(' ')
                          PSASTWrapperElement(TypeAtom)
                            PSASTWrapperElement(TypeConstructor)
                              PSASTWrapperElement(Qualified)
                                PSASTWrapperElement(TypeConstructor)
                                  PsiElement(proper name)('User')
                        PsiElement())(')')
                  PsiElement())(')')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('age')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('b')
                  PsiElement())(')')
                  PsiElement(whitespace)('   ')
            PsiElement(})('}')