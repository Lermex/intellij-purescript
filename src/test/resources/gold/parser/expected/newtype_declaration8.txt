Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
        PsiElement(PSTokenType.newtype)('newtype')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('Cool')
        PSIdListImpl(ID_LIST)
          PSDeclarationIdImpl(DECLARATION_ID)
            PsiElement(PSTokenType.ID)('a')
          PSIdListImpl(ID_LIST)
            PSDeclarationIdImpl(DECLARATION_ID)
              PsiElement(PSTokenType.ID)('b')
            PSIdListImpl(ID_LIST)
              PSDeclarationIdImpl(DECLARATION_ID)
                PsiElement(PSTokenType.ID)('c')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
          PSTypeListImpl(TYPE_LIST)
            PSDeclarationTypeImpl(DECLARATION_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('Cool')
          PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
            PsiElement(PSTokenType.{)('{')
            PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
              PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                PsiElement(PSTokenType.ID)('name')
                PsiElement(PSTokenType.::)('::')
                PsiElement(PSTokenType.()('(')
                PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                  PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('String')
                PsiElement(PSTokenType.->)('->')
                PsiElement(PSTokenType.()('(')
                PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                  PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('Array')
                    PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                      PSDeclarationTypeImpl(DECLARATION_TYPE)
                        PsiElement(PSTokenType.PROPER_NAME)('User')
                      PsiElement(PSTokenType.))(')')
                  PsiElement(PSTokenType.))(')')
              PsiElement(PSTokenType.,)(',')
              PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
                PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                  PsiElement(PSTokenType.ID)('age')
                  PsiElement(PSTokenType.::)('::')
                  PsiElement(PSTokenType.()('(')
                  PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('a')
                  PsiElement(PSTokenType.->)('->')
                  PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('b')
                    PsiElement(PSTokenType.))(')')
            PsiElement(PSTokenType.})('}')